### PR TITLE
VIH-7149 Interpreter-Interpretee UI Link

### PR DESCRIPTION
**Pull request by williambowshersolirius**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-7149

### Change description ###
Make sure that the Intrepretee always immediately precedes their Interpreter in the participant list when either a Judge or a  Participant is in the waiting room 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

**Review by levibutler — APPROVED** [review]

**Comment by levibutler** [comment]
For the future, if have to merge master or rebase if we could keep the same PR so we can see the history of the comments, please

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts [hunk 1]
```diff
@@ -254,4 +255,121 @@ describe('WaitingRoom ParticipantList Base', () => {
 
         expect(component.canInvite).toBe(true);
     });
-});
+
+    describe('JudgeParticipantStatusListComponentWithInterpreter', () => {
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts:259]: Can you move tests for judge participant status lists in the correct place? But by the looks of the test, the description isn't correct

Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts:259]: This describe looks like a copy and paste of the previous describe. What new scenarios are you actually adding?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts [hunk 2]
```diff
@@ -104,9 +105,54 @@ export abstract class WRParticipantStatusListDirective {
     }
 
     protected filterNonJudgeParticipants(): void {
-        this.nonJudgeParticipants = this.conference.participants.filter(
+        const nonJudgeParts = this.conference.participants.filter(
             x => x.role !== Role.Judge && x.role !== Role.JudicialOfficeHolder && x.hearing_role !== HearingRole.OBSERVER
         );
+
+        const individualWithInterpreter = nonJudgeParts.find(
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:112]: This finds the first individual with an interpreter, what happens if you have multiple interpreters?

Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:112]: Would a view model not be best suited here like the judge participant panel model?

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
Not sure what the additional describe is providing. Also, I'm not sure this implementation will handle sorting hearings with multiple interpreters

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts:259]: This looks to be testing functions in the base class `WRParticipantStatusListDirective` and nothing to do functions specific to `JudgeParticipantStatusListComponent`

**Review by shaed-parkar — CHANGES_REQUESTED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts (hunk 1 above)
Comment (williambowshersolirius) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts:259]: I am checking that the Intrepretee precedes the Interpreter in the participant list and that the refactored UI elements display the correct text based on the Hearing role

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts]: lets just rename the test here and below :)

Comment (williambowshersolirius) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.spec.ts]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts (hunk 2 above)
Comment (williambowshersolirius) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts]: done

**Review by shaed-parkar — APPROVED** [review]
LGTM

**Pull request merged**